File: src/utils/application_event_filter.cpp
#include "application_event_filter.h"
#include <QEvent>
#include <QApplicationStateChangeEvent>
#include <QDebug>

bool ApplicationEventFilter::eventFilter(QObject *obj, QEvent *event) {
    // qDebug() << event->type();

    auto event_type = event->type();

#ifdef Q_OS_MAC
    if (event_type == QEvent::ApplicationStateChange) {
        Qt::ApplicationState state = static_cast<QApplicationStateChangeEvent*>(event)->applicationState();
        if (state == Qt::ApplicationActive) {
            emit appEvent("MAC_ApplicationActive", QVariant());
        }
    }
    else if (event_type == QEvent::Quit) {
        emit appEvent("MAC_Quit", QVariant());
    }
#endif

    return QObject::eventFilter(obj, event);
}

File: src/utils/application_event_filter.py
import platform
from PySide6.QtCore import Qt, QObject, Signal, QEvent
from PySide6.QtWidgets import QApplication

class ApplicationEventFilter(QObject):
    appEvent = Signal(str, 'QVariant')

    def eventFilter(self, obj: QObject, event: QEvent):
        event_type = event.type()

        if platform.system() == 'Darwin':
            if event_type == QEvent.ApplicationStateChange:
                state = QApplication.instance().applicationState()
                if state == Qt.ApplicationActive:
                    self.appEvent.emit("MAC_ApplicationActive", {})
            elif event_type == QEvent.Quit:
                state = QApplication.instance().applicationState()
                self.appEvent.emit("MAC_Quit", {})

        return super().eventFilter(obj, event)
